modules/extension/xsd/xsd-core/src/main/java/org/geotools/xs/bindings/XSAnyTypeBinding.java

```diff
@@ -115,7 +115,8 @@ public Object parse(ElementInstance instance, Node node, Object value)
         String text = null;
 
         if ((value != null) && value instanceof String) {
-            text = ((String) value).trim();
+          //GEOS-8227 don't trim text it will already be trimmed unless in CDATA
```

Comment:
I'd just remove these comments, people can use history to check what happened and tests should protect the code